src/query_plan/query_planning_traversal.rs

```diff
@@ -427,10 +429,58 @@ impl<'a> QueryPlanningTraversal<'a> {
 
     fn selection_set_is_fully_local_from_all_nodes(
         &self,
-        _selection: &NormalizedSelectionSet,
-        _nodes: &IndexSet<NodeIndex>,
+        selection: &NormalizedSelectionSet,
+        nodes: &IndexSet<NodeIndex>,
     ) -> Result<bool, FederationError> {
-        todo!()
+        // To guarantee that the selection is fully local from the provided vertex/type, we must have:
```

Comment:
looks like we are missing this [block](https://github.com/apollographql/federation/blob/main/query-planner-js/src/buildPlan.ts#L257-L265) which checks for inconsistent abstract types on fragment conditions?